src/transaction/log_recovery.c

```diff
@@ -1801,8 +1801,8 @@ log_rv_analysis_complete (THREAD_ENTRY * thread_p, int tran_id, LOG_LSA * log_ls
 	  // newer quick fix on top of older quick fix: mark the mvccid as completed
 	  log_rv_analysis_complete_mvccid (tran_index, tdes);
 
-	  // quick fix: reset mvccid.
-	  tdes->mvccinfo.id = MVCCID_NULL;
+	  // quick fix: reset mvcc.
+	  tdes->mvccinfo.reset ();
```

Comment:
How about putting `tdes->mvccinfo.reset ();` into `logtb_clear_tdes()` and removing this line?

Reply:
I think it is better to put `tdes->mvccinfo.reset ()` into `logtb_clear_tdes ()`. 

`logtb_clear_tdes ()`, I think, is designed to be called after `logtb_complete_mvcc ()`. 
because mvcc related information is only set in the active transaction, and will be cleared after transaction is completed.

```
log_commit/abort () 
-> log_commit/abort_local () //  calls logtb_complete_mvcc () to reset mvccinfo
-> log_complete ()  // calls logtb_clear_tdes ()
```

However, recovery is not considered.
In `log_rv_undo_abort_complete()` and `log_rv_analysis_complete()', 
it just set mvccid to null and clear the tdes. 
Setting mvccid to null (`mvccinfo.reset ()`) is what logtb_complete_mvcc() was doing (which is called in log_commit_local() or log_abort_local ()). 

So, last_mvcc_lsa is not reset during recovery, and this will make other problem.

Reply:
I think I understand what you mean:
- on the path `log_rv_analysis_complete` -> `logtb_free_tran_index` -> `logtb_clear_tdes`, in function `logtb_clear_tdes` there is a pre-condition that `assert (tdes->mvccinfo.id == MVCCID_NULL)` which I would like us to keep.
- so you say that re-setting the `mvccid` is still needed here - and I agree
- so, let's leave just the `mvccinfo.id = null` as before and still add in the `logdb_clear_tdes` - as I suggested in a previous comment :
```
assert (tdes->mvccinfo.id == MVCCID_NULL);
tdes->mvccinfo.reset ();
```

A similar thing happens in `log_rv_undo_abort_complete`:
- on the path `log_rv_undo_abort_complete` -> `log_complete` -> `logtb_clear_tdes`
- the same precondition from `logtb_clear_tdes` mandates leaving the line
```
  tdes->mvccinfo.id = MVCCID_NULL;
```
in `log_rv_undo_abort_complete`

Reply:
What I meant to say was to remove the part that initializes mvccid (`tdes->mvccinfo.id = MVCCID_NULL;`) inside the `log_rv_analysis_complete` and `log_rv_undo_abort_complete` functions.
This part looks like a quick fix to pass the `assert (tdes->mvccinfo.id == MVCCID_NULL);`,
and the above assert statement doesn't seem to take the recovery process into account, so I think it seems unnecessary. 

my opinion : 
1) remove `tdes->mvccinfo.id = MVCCID_NULL;` in `log_rv_analysis_complete` and `log_rv_undo_abort_complete`
2) remove `assert (tdes->mvccinfo.id == MVCCID_NULL);` in `logtb_clear_tdes ()`
3) put `mvccinfo.reset ()` in `log_clear_tdes ()`

But I also understand your opinion of reducing risk. First of all, I will modify according to your opinion. After further analysis, we will proceed with refactoring later.